graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +46,392 @@ def __init__(self, query_ast):
         # List[SubQueryNode], the queries that depend on the current query
 
 
+def _split_query_ast_one_level_recursive(
+    query_node, ast, type_info, edge_to_stitch_fields, intermediate_out_name_assigner
+):
+    """Return an AST node with which to replace the input AST in the selections that contain it.
+
+    This function examines the selections of the input AST, and divides them into three sets:
+    property fields, intra-schema vertex fields, and cross-schema vertex fields.
+
+    Each cross-schema vertex field will have its branch removed from the output AST, and made
+    into its own separate query AST. The parent and child property fields used in
+    the stitch will be added to the parent and child ASTs. @output directives will be added
+    to the parent and child property fields, if not already present.
+
+    The function will be called recursively on each intra-schema vertex field.
+
+    Args:
+        query_node: SubQueryNode, whose list of child query connections may be modified to
+                    include new children
+        ast: Field, InlineFragment, or OperationDefinition, the AST that we are trying to split
+             into child components. It is not modified by this function
+        type_info: TypeInfo, used to get information about the types of fields while traversing
+                   the query ast
+        edge_to_stitch_fields: Dict[Tuple(str, str), Tuple(str, str)], mapping
+                               (type name, vertex field name) to
+                               (source field name, sink field name) used in the @stitch directive
+                               for each cross schema edge
+        intermediate_out_name_assigner: IntermediateOutNameAssigner, object used to generate
+                                        and keep track of names of newly created @output
+                                        directives
+    """
+    # Split selections into three kinds: property fields, cross schema vertex fields, normal
+    # vertex fields/type coercions
```

Comment:
You're right, I didn't do a special case before because I thought it's possible to have type coercions coexist with other fields and type coercions. Will do a special case now.